crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -797,6 +827,46 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.types.definitions.insert(definition, context_expr_ty);
     }
 
+    /// Infer the types of exceptions caught by an exception handler
+    fn infer_handled_exception_types<'a, 'b>(
```

Comment:
You could get away without introducing your own iterator type by using a slice iterator instead

```rust
    fn infer_handled_exception_types(
        &self,
        handled_exceptions: &'db ast::Expr,
    ) -> impl Iterator<Item = ExceptHandlerType<'db>> + '_ {
        let exceptions = match handled_exceptions {
            ast::Expr::Tuple(multiple_exceptions) => multiple_exceptions.elts.as_slice(),
            single_exception => std::slice::from_ref(single_exception),
        };

        exceptions.into_iter().map(move |exception| {
            let node_ty = self
                .types
                .expression_ty(exception.scoped_ast_id(self.db, self.scope));

            let ty = match node_ty {
                Type::Class(exception_class) => ExceptHandlerType::Valid {
                    exception_ty: Type::Instance(exception_class),
                },
                Type::Any | Type::Unknown => ExceptHandlerType::Valid {
                    exception_ty: node_ty,
                },
                invalid_ty => ExceptHandlerType::Invalid {
                    node: exception,
                    node_ty: invalid_ty,
                },
            };

            ty
        })
    }
```

Reply:
Thanks, TIL!

Reply:
`std::slice::from_ref` is awesome :)